lib/tcp_connection.cpp

```diff
@@ -142,7 +142,7 @@ TCPConnection::parse_address(std::string& addr, unsigned int& port,
 	// Ensure that there are only alphanumeric characters, dots,
 	// dashes and colons in address.  Anything else must be an error.
 #if __cplusplus >= 201103L
-	for (char c : addr) {
+	for (const char& c : addr) {
```

Comment:
This is simply wrong. You're exchanging a copy of a 1-byte value with a copy of a `sizeof(void*)` pointer (the implementation underpinning the reference) and then you're dereferencing that pointer in each iteration of the loop.

[Study the differences on Compiler Explorer](https://godbolt.org/z/3WfxfK).

Notice that cases A and B (my original and your first attempt) are basically identical, showing that the "enhancement" this patch is based on doesn't actually solve anything, other than by making the code a bit shorter, ironically by making it longer!

But then notice that from B to C (this patch's "improvement") the assembly's gotten longer, particularly inside the actual loop!

You've made it slower, not faster.

Reply:
Yeah, my stupid mistake, sorry.